src/validationinterface.h

```diff
@@ -35,6 +36,8 @@ class CValidationInterface {
     virtual void Inventory(const uint256 &hash) {};
     virtual void ResendWalletTransactions(int64_t nBestBlockTime) {};
     virtual void BlockChecked(const CBlock&, const CValidationState&) {};
+    virtual void GetScriptForMining(CScript &script) {};
```

Comment:
I think the right solution for that is having a trivial not-really-a-wallet
provider that explicitly allows you to set the script output to mine to. It
just looks incorrect to have a provider interface for something, and a
default implementation that results in the least expected result.